crates/fs/src/repository.rs

```diff
@@ -119,7 +117,7 @@ impl GitRepository for LibGitRepository {
 
         if let Some(statuses) = self.statuses(Some(&mut options)).log_err() {
             for status in statuses.iter() {
-                let path = RepoPath(PathBuf::from(OsStr::from_bytes(status.path_bytes())));
+                let path = RepoPath(PathBuf::try_from_bytes(status.path_bytes()).unwrap());
```

Comment:
It's not great that we're losing the infallible conversion on *nix targets.

Seems like we should use conditional compilation here to do a specific conversion only on Windows and leave the existing conversion to `PathBuf` as-is for macOS and Linux.

Reply:
This function, [which I added previously to absorb platform dependencies](https://github.com/zed-industries/zed/pull/7072), returns a Result, but is equivalent to the existing conversion on Linux and macOS, so the actual panic may only occur on Windows.